vaadin-notification-flow-parent/vaadin-notification-flow/src/main/java/com/vaadin/flow/component/notification/Notification.java

```diff
@@ -433,12 +445,14 @@ public void setOpened(boolean opened) {
                     + "That may happen if you call the method from the custom thread without "
                     + "'UI::access' or from tests without proper initialization.");
         }
-        if (opened && getElement().getNode().getParent() == null) {
-            ui.beforeClientResponse(ui, context -> {
+
+        ui.beforeClientResponse(ui, context -> {
+            if (opened && getElement().getNode().getParent() == null) {
```

Comment:
Should this use `getOpened`? `opened` might be stale at this point.

Reply:
Done